Step-80

luca-heltai opened the pull request:
I'm adding here what I found on my pc for step-80. @fdrmrc, @blaisb, this is where we currently are. I think we are not too far away with the functionality. @mwichro asked recently about the status of this.

This is now on our to-do-list for dealii-X.

luca-heltai committed:
Coupling between non matching parallel distributed objects

Handle all things needed to couple non matching parallel distributed
triangulations through particle handler objects.

Co-authored-by: Bruno Blais <blais.bruno@gmail.com>

luca-heltai committed:
Fresh start = step-70

luca-heltai committed:
Starting point.

luca-heltai committed:
Small changes.

blaisb committed:
Update CMakeList

blaisb committed:
Particles + solid matrix

luca-heltai committed:
Assemble coupling

blaisb committed:
Fix properties + mask

luca-heltai committed:
DSP.

luca-heltai committed:
Initialize coupling.

luca-heltai committed:
AA op.

luca-heltai committed:
Generated ops.

luca-heltai committed:
Fixed compilation warnings with 9.8.pre.

luca-heltai committed:
Added builds-on

luca-heltai committed:
Make indent happy.

luca-heltai committed:
Running.

blaisb commented:
@luca-heltai I think this calls for a visit to Pisa :P

luca-heltai commented:
> @luca-heltai I think this calls for a visit to Pisa :P

That's the idea... :)

luca-heltai committed:
Stokes -> NavierStokes

luca-heltai committed:
Navier Stokes working (?)

bangerth reviewed (commented):
Let us know when it's in a state where you want others to review it!

luca-heltai committed:
new version.

fdrmrc committed:
Debug: lid drive with immersed ball

luca-heltai committed:
Working rotating ball.

luca-heltai committed:
Start documenting.

luca-heltai committed:
Fix doxygen.

luca-heltai committed:
rhs.

luca-heltai committed:
Fix compilation.

luca-heltai committed:
Fix names.

fdrmrc committed:
Uncomment Timers

fdrmrc committed:
Fix std::bad_alloc error

fdrmrc committed:
Fix wrong rhs update

luca-heltai committed:
Renamed variables.

fdrmrc committed:
Workish AL preconditioner. To be cleaned and refined

fdrmrc committed:
Make AL parameter depend on Delta_t. Good iteration counts, yet to be cleaned

luca-heltai committed:
Fix AL inversions.

luca-heltai committed:
WORK IN PROGRESS. DEBUG SPACES.

luca-heltai committed:
change pos.

luca-heltai committed:
Flowing in the wrong direction??

luca-heltai committed:
Small improvements.

luca-heltai committed:
Minor improvements. Not working.

luca-heltai committed:
Working with Q2-Q1-Q1-Q1.

luca-heltai committed:
Working in parallel Q2-Q1-Q1-Q1

luca-heltai committed:
DG spaces?

luca-heltai committed:
Working for different spaces.

luca-heltai committed:
Make sure we can use different spaces.

luca-heltai committed:
High order output.

blaisb committed:
Add generation of the output folder if it does not exist

luca-heltai committed:
Added TODO list. Going for lucnh.

blaisb committed:
Indent and fix default parameters

luca-heltai committed:
Update todo.

luca-heltai committed:
Move AMG initialization.

blaisb committed:
Add tracer particles

blaisb committed:
Update TODO now that tracer is done

blaisb committed:
Update todo

luca-heltai committed:
Compile and run with both Trilinos and PETSc.

luca-heltai committed:
Fix todos.

blaisb commented:
TODO list for step-80:

- [x] Create output directory if it does not exist
- [x] Functions should be initialized correctly
- [x] Default parameter should work
- [x] Move initialization of AMG preconditioners to the assembly functions
- [x] Reinsert tracers as in step-70
- [x] Allow the use of MUMPS for linear solver
- [x] Make sure MUMPS work (not our issue here)
- [x] Read second argument from CLI and make sure to save used parameter file there
- [x] Assemble the augmented system directly, and use AMG on it
- [x] Make sure we use the same format for the parameters that is used as input (i.e., json/prm)
- [x] Add output with only changed parameters w.r.t. default
- [x] Make it robust w.r.t. going out of the domain
- [x] Recover from disaster (e.g., going out of the domain)
- [x] Fix one dof of pressure if all bc are Dirichlet
- [x] Use FEPointEvaluator for particle interpolation operator
- [X] Make sure we output the correct initial condition --BB
- [X] Time dependent boundary conditions --BB
- [x] Add statistics on the number of inner iterations, and number of iterations for the augmented system
- [x] Remove MUMPs part once iterative solver is robust (or move it to another PR)
- [x] Remove tracer particle handler
- [x] Enable semi-explicit nonlinear elasticity instead of linear elasticity
- [ ] Try operator preconditioning? ---- future work
- [ ] Explain very well the choices of the reference configuration
- [ ] Use other pull requests (DofHandlerCoupling) - BB
- [x] Remove (maybe?) solid particle handler, only use DoFHandlerCoupling
- [ ] Initial solid partitioning? --- Move to possibility for extension
- [x] <s>Make sure we work with fullydistributed and simplex grids.</s>
- [x] <s>Use Sundials IDA</s>
- [x] Oscillating Leaftlet
- [ ] Turek & Hron FSI example FSI-3 
- [x] Converging - diverging nozzle + capsule

luca-heltai committed:
Concept for time adaptivity.

luca-heltai committed:
Removed todo.

fdrmrc committed:
Fix dangling reference in lambda

fdrmrc committed:
Allow grad-div stabilization. Fix couple of warnings

luca-heltai committed:
Add IDA support

blaisb committed:
Add particle predictor for the object location

luca-heltai committed:
NOT WORKING. RM ME.

fdrmrc committed:
WIP: add AL term using particles

luca-heltai committed:
Fix compilation and changed outputs.

luca-heltai committed:
Fixed parallel codes.

fdrmrc committed:
Build augmented term in one shot

luca-heltai committed:
Fix header.

blaisb committed:
Fix parallel and things work

luca-heltai committed:
Remove unused typedef.

fdrmrc committed:
Fix name for augmentation parameter

blaisb committed:
Rename the tests to prepare addition of others

luca-heltai committed:
Integrate vector + tests.

luca-heltai committed:
Try fix. not working yet.

luca-heltai committed:
Refactor.

luca-heltai committed:
Added constraints.

luca-heltai committed:
Fix with trilinos vectors.

blaisb committed:
Add a working test for deal.II distributed vectors

blaisb committed:
Add some slight comments

blaisb committed:
Rename test 01 that requires triliinos

blaisb committed:
Indent and add test 04

blaisb committed:
Add more comments for test 04

blaisb committed:
Make sure tests require Trilinos

blaisb committed:
Allow compilation in codim=2 + Mass matrix

blaisb committed:
Indent

blaisb committed:
Add capability to assemble transpose or not

Checkpoint before moving the constraints to the constructor

blaisb committed:
Fix constraint positions correctly

blaisb committed:
Rename create_coupling_mass_sparsity_pattern

fdrmrc committed:
Add TODO for MUMPS solver

luca-heltai committed:
Refactor coupling matrix.

blaisb committed:
Fix the initial condition to be working correctly

blaisb committed:
Constant mode Navier-Stokes and fix RHS for operator augmentation

The operator augmentation was missing from the RHS which
prevent translation cases from workig correctly. This commit fixes
this by correctly adding the right term!

blaisb committed:
Separate the solid_preconditioner

blaisb committed:
rename 1./timestep to alpha and change API

blaisb committed:
Alpha change

luca-heltai committed:
Try fixing residual.

luca-heltai committed:
Improve dof handler coupling apis.

luca-heltai committed:
Fix compilation.

luca-heltai committed:
Try IDA.

blaisb committed:
Fix cell_preconditioner

luca-heltai committed:
RM me.

luca-heltai committed:
Add gtest.

blaisb committed:
Add analytical solution + NS dedicated solver+MMS

luca-heltai committed:
Indent

fdrmrc committed:
Add elasticity example

luca-heltai committed:
Testing the coupling.

fdrmrc committed:
Quad degree

luca-heltai committed:
Remove constraints from coupling.

fdrmrc committed:
Simplify model, check for bad mappings, remove IDA path

fdrmrc committed:
Removed IDA

fdrmrc committed:
Add parameters file for square test case

fdrmrc committed:
Add non linear model with Netwon solver

blaisb committed:
Upgrade preconditioner

luca-heltai committed:
Working scaling and parameters.
